Assets/Scripts/Player/PlayerView.cs

```diff
@@ -32,21 +36,23 @@ public void MoveTank(float hVal, float vVal, float speed, float rotateSpeed)
 
         public void Shoot(BulletController bulletController)
         {
-            bulletController.SpawnBullet(transform.forward, bulletSpawnPos.transform.position, transform.eulerAngles);
+            bulletController.SpawnBullet(transform.forward, bulletSpawnPos.transform.position, transform.eulerAngles, playerController);
         }
 
         private void OnCollisionEnter(Collision collision)
         {
             if (collision.gameObject.tag == "Enemy")
             {
-                PlayerManager.Instance.DestroyPlayer(playerController);
-
+                playerHealth -= playerHealth;
```

Comment:
health calculations should be in the controller - all logic inside Controller

Reply:
Done.